src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -1,264 +1,285 @@
+            }
+            finally
+            {
+                TelemetryManager.Log.WriteEvent(commandEvent);
+            }
+        }
+
+        /// <summary>
+        /// Convert list of yaml contents to Manifests object model, and then update them.
+        /// </summary>
+        /// <param name="latestManifestContent">List of manifests to be updated.</param>
+        /// <returns>Manifests object representing the updates manifest content, or null if the update failed.</returns>
+        public async Task<Manifests> DeserializeExistingManifestsAndUpdate(List<string> latestManifestContent)
+        {
+            Manifests manifests = new Manifests();
+
+            Serialization.DeserializeManifestContents(latestManifestContent, manifests);
+
+            if (manifests.SingletonManifest != null)
+            {
+                manifests = ConvertSingletonToMultifileManifest(manifests.SingletonManifest);
+            }
+
+            VersionManifest versionManifest = manifests.VersionManifest;
+            InstallerManifest installerManifest = manifests.InstallerManifest;
+            DefaultLocaleManifest defaultLocaleManifest = manifests.DefaultLocaleManifest;
+            List<LocaleManifest> localeManifests = manifests.LocaleManifests;
+
+            // Ensure that capitalization matches between folder structure and package ID
+            versionManifest.PackageIdentifier = this.Id;
+            installerManifest.PackageIdentifier = this.Id;
+            defaultLocaleManifest.PackageIdentifier = this.Id;
+            UpdatePropertyForLocaleManifests(nameof(LocaleManifest.PackageIdentifier), this.Id, localeManifests);
+
+            if (!string.IsNullOrEmpty(this.Version))
+            {
+                versionManifest.PackageVersion = this.Version;
+                installerManifest.PackageVersion = this.Version;
+                defaultLocaleManifest.PackageVersion = this.Version;
+                UpdatePropertyForLocaleManifests(nameof(LocaleManifest.PackageVersion), this.Version, localeManifests);
+            }
+
+            if (!this.InstallerUrls.Any())
+            {
+                this.InstallerUrls = installerManifest.Installers.Select(i => i.InstallerUrl).Distinct().ToArray();
+            }
+
+            // We only support updates with same number of installer URLs
+            if (this.InstallerUrls.Distinct().Count() != installerManifest.Installers.Select(i => i.InstallerUrl).Distinct().Count())
+            {
+                Logger.ErrorLocalized(nameof(Resources.MultipleInstallerUpdateDiscrepancy_Error));
+                return null;
+            }
+
+            foreach (var url in this.InstallerUrls)
+            {
+                string packageFile = await DownloadPackageFile(url);
+                if (string.IsNullOrEmpty(packageFile))
+                {
+                    return null;
+                }
+
+                this.packageFiles.Add(packageFile);
+            }
+
```

Comment:
I think this block of code is almost identical to the one in the NewCommand. You can refactor and put it in the basecommand class.